pkg/scheduler/framework/plugins/noderesources/balanced_allocation.go

```diff
@@ -67,13 +65,15 @@ func (ba *BalancedAllocation) ScoreExtensions() framework.ScoreExtensions {
 }
 
 // NewBalancedAllocation initializes a new plugin and returns it.
-func NewBalancedAllocation(_ runtime.Object, h framework.Handle) (framework.Plugin, error) {
+func NewBalancedAllocation(_ runtime.Object, h framework.Handle, enableBalanceAttachedNodeVolumes, enablePodOverhead bool) (framework.Plugin, error) {
```

Comment:
As we discussed in https://github.com/kubernetes/kubernetes/pull/98545#issuecomment-770056695, prefer to use a map or WithXYZOption pattern as the input params. Otherwise, once some features reach to GA, the signature of this function needs to be updated.

Reply:
Yeah, I like the `WithXYZOption` approach, something like `.WithFeaturegates(map[featuregate.Feature]bool)`?

Reply:
Are we going to have the featuregate package in staging?

What if we do: `WithEnableXYZ`, where xyz is the specific feature, so each plugin have an explicit flag that gets set.

Reply:
@Huang-Wei actually pointed out that `featuregate.Feature` is from component-base already: https://github.com/kubernetes/component-base/blob/1722fee/featuregate/feature_gate.go#L33. But the feature name constants are still in `pkg/features` so I don't know if that would work.

Using `WithEnableXYZFeature` would probably be easier to work with

Reply:
do we still want to use the With Pattern (without changing the PluginFactory signature) just so we avoid mixing the flags (values passed to a series of bool parameters can easily be switched mistakenly)

Reply:
+1
Make sure the `WithMyFeatureFooEnabled` functions belong to each plugin package.